### DISCUSSION TITLE
Performance downgraded for larger number of json

### DISCUSSION THREAD

**Discussion by zliucd** [nlohmann/json#3372] [General]
Hi,

There is a major performance bottleneck when using ```nlohmann:json```. In my scenario, there are 300 hundred JSON objects, and each JSON has 1,000 keys.  After serializing using ```json```, the performance is downgraded by 8-10x times. 

 I tried ordered_json and json, and results are identical.  I also use Valgrind for performance profiling, and ```callgrind``` results indicate ```json``` incurs high overhead.     As ```json``` and ```ordered_json``` are built on top of ```STL```,  so the real cause is STL?

How to boost the performance? Appreciate your help.

**Comment by nlohmann** [comment]
Did you compile in Release mode?

**Reply by zliucd** [reply]
@nlohmann  Thanks a lot. I just tested in Release mode, and performance improved significantly(from 8-10x times overhead to 2x), which is a very good result :)
